cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/event/EventBus.java

```diff
@@ -88,9 +107,21 @@ public void register(Object object) {
         do {
             for (Method m : listenerClass.getDeclaredMethods()) {
                 if (m.isAnnotationPresent(com.cloudant.sync.event.Subscribe.class)) {
+
+                    if (!Modifier.isPublic(m.getModifiers())){
+                        throw new IllegalArgumentException(String.format(Locale.ENGLISH,
+                                "Subscriber method %s#%s is required to be public",
```

Comment:
No they can't. Anonymous will always be rejected by the first check because they cannot be made public.